components/automate-ui/src/app/pages/project/rules/project-rules.component.ts

```diff
@@ -76,55 +76,87 @@ export class ProjectRulesComponent implements OnInit, OnDestroy {
     private route: ActivatedRoute,
     private fb: FormBuilder) {
 
-      combineLatest([
-        this.store.select(getStatus),
-        this.store.select(updateStatus),
-        this.store.select(getProjectStatus)
-      ]).pipe(
-        takeUntil(this.isDestroyed)
-      ).subscribe(([gStatus, uStatus, gpStatus]) => {
-        const routeId = this.route.snapshot.paramMap.get('ruleid');
-        this.isLoading = routeId &&
-          (gStatus !== EntityStatus.loadingSuccess) ||
-          (uStatus === EntityStatus.loading) ||
-          (gpStatus !== EntityStatus.loadingSuccess);
-        });
-
-      combineLatest([
-        this.store.select(routeParams).pipe(pluck('id'), filter(identity)),
-        this.store.select(routeParams).pipe(pluck('ruleid'), filter(identity))
-      ]).pipe(
-        takeUntil(this.isDestroyed)
-      ).subscribe(([project_id, rule_id]: string[]) => {
-        this.store.dispatch(new GetProject({ id: project_id }));
-        this.store.dispatch(new GetRule({
-          id: rule_id,
-          project_id
-        }));
+    combineLatest([
```

Comment:
Way are all these in the constructor and not in the "ngOnInit"?

Reply:
We could also change the `this.isLoading` to an observable<boolean>. Then we do not have to subscribe, like in the below branch. 
https://github.com/chef/automate/compare/bhd/2179/project-subscriptions...lancewf/option

Reply:
good idea! shouldn't take too long to implement